Lib/test/test_ntpath.py

```diff
@@ -866,18 +866,25 @@ def test_commonpath(self):
         def check(paths, expected):
             tester(('ntpath.commonpath(%r)' % paths).replace('\\\\', '\\'),
                    expected)
-        def check_error(exc, paths):
-            self.assertRaises(exc, ntpath.commonpath, paths)
-            self.assertRaises(exc, ntpath.commonpath,
+        def check_error(exc, exp, paths):
+            self.assertRaisesRegex(exc, exp, ntpath.commonpath, paths)
+            self.assertRaisesRegex(exc, exp, ntpath.commonpath,
                               [os.fsencode(p) for p in paths])
 
         self.assertRaises(TypeError, ntpath.commonpath, None)
         self.assertRaises(ValueError, ntpath.commonpath, [])
         self.assertRaises(ValueError, ntpath.commonpath, iter([]))
-        check_error(ValueError, ['C:\\Program Files', 'Program Files'])
-        check_error(ValueError, ['C:\\Program Files', 'C:Program Files'])
-        check_error(ValueError, ['\\Program Files', 'Program Files'])
-        check_error(ValueError, ['Program Files', 'C:\\Program Files'])
+
+        # gh-117381: Logical error messages, feel free to change drive error
+        # to mix error
+        check_error(ValueError, "Paths don't have the same drive", ['C:\\Program Files', '\\Program Files'])
```

Comment:
For all tests, also check the arguments in a different order. The error message should be the same.

The patch can have a drive or not, be rooted or not, so there are 4 different kinds of paths. It gives 16 combination of pairs, 4 of them are valid, -- there should be 12 error tests.

Reply:
I've updated `check_error` to also test the reversed list. I have a schema of all invalid pairs above.